libraries/aws_ecr_repository_policy.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+require 'aws_backend'
+
+class AwsEcrPolicy < AwsResourceBase
+  name 'aws_ecr_repository_policy'
+  desc 'Verifies policy settings for an AWS ECR Repository'
+
+  example "
+    describe aws_ecr_repository_policy(repository_name: 'my-repository') do
+      it { should exist }
+    end
+  "
+  attr_reader :repository_name
+
+  def initialize(opts = {})
+    # Create a repository_name:<value> pair if the argument is a string object.
+    opts = { repository_name: opts } if opts.is_a?(String)
+    # Ignore arguments if they are not hash type at this point.
+    opts = {} unless opts.is_a?(Hash)
```

Comment:
this is not required! as `validate_parameters` implicitly checks if opts is of Hash type and raises `ArgumentError`

Reply:
modified